### PR TITLE
custom relations bugfix

### PR DESCRIPTION
**Pull request by mvallotto**

### CODE REVIEW

File: js-data-memory.js [hunk 1]
```diff
@@ -153,18 +156,20 @@ function MemoryAdapter(opts) {
...
    */
-  this._update = function (resource, id, attrs) {
+  this._update = function (resource, id, attrs, options) {
     addMetaForResource(resource);
-    return this.find(resource, id).then(function (item) {
+    var self = this;
+    return this.find(resource, id, options).then(function (item) {
       if (attrs) {
-        Object.keys(attrs).forEach(function(key) {
+        for (var key in attrs) {
```
Comment (finwo) [inline js-data-memory.js]: No `.hasOwnProperty` check?

Comment (finwo) [reply js-data-memory.js]: Declaring a var in a loop statement is something I always try to prevent.

File: js-data-memory.js [hunk 2]
```diff
@@ -269,10 +271,10 @@ function MemoryAdapter(opts) {
    * @param {*} mapper 
    * @param {*} def 
    * @param {*} records 
-   * @param {*} __opts 
+   * @param {*} ___opts
```
Comment (finwo) [inline js-data-memory.js]: You might have 1 too many underscores.

File: js-data-memory.js [hunk 3]
```diff
@@ -336,10 +338,10 @@ function MemoryAdapter(opts) {
...
```
Comment (finwo) [inline js-data-memory.js]: You might have 1 too many underscores.

File: js-data-memory.js [hunk 4]
```diff
@@ -451,41 +464,53 @@ function MemoryAdapter(opts) {
...
```
Comment (finwo) [inline js-data-memory.js]: You might have 1 too many underscores.

File: js-data-memory.js [hunk 5]
```diff
@@ -451,41 +464,53 @@ function MemoryAdapter(opts) {
...
-   * @param {*} __opts 
+   * @param {*} ___opts 
    */
   this.loadHasManyForeignKeys = function(mapper, def, records, __opts) {
-    var _this12 = this;
+    var self = this;
 
     var relatedMapper = def.getRelation();
-    var record        = false;
+    var idAttribute   = mapper.idAttribute;
```
Comment (finwo) [inline js-data-memory.js]: You never use this. Can it be removed?

File: js-data-memory.js [hunk 6]
```diff
@@ -451,41 +464,53 @@ function MemoryAdapter(opts) {
...
+        records.forEach(function (record) {
+          p.push(self.findAll(relatedMapper, {
+            where : defineProperty({}, def.foreignKeys, {
+              'contains' : self.makeHasManyForeignKeys(mapper, def, record)
+            })
+          }, __opts));
+        })
+        return Promise.all(p)
+        .then(function (relatedItems) {
+          for ( var i in records ) {
```
Comment (finwo) [inline js-data-memory.js]: No `.hasOwnProperty` check?

Comment (finwo) [reply js-data-memory.js]: I always try to avoid declaring a var in a for declaration this way.

File: js-data-memory.js [hunk 7]
```diff
@@ -495,22 +520,34 @@ function MemoryAdapter(opts) {
    * @param {*} mapper 
    * @param {*} def 
    * @param {*} records 
-   * @param {*} __opts 
+   * @param {*} ___opts
```
Comment (finwo) [inline js-data-memory.js]: You may have 1 too many underscores.

**Comment by mvallotto** [comment]
everything edited

**Pull request merged**